src/main/java/io/confluent/copycat/hdfs/hive/HiveExec.java

```diff
@@ -0,0 +1,135 @@
+  private static final String HIVE_SASL_ENABLED = "hive.metastore.sasl.enabled";
+
+  public HiveExec(Configuration conf, HdfsSinkConnectorConfig config) {
+    this.conf = conf;
+    String topicsDir = config.getString(HdfsSinkConnectorConfig.TOPIC_DIR_CONFIG);
+    String url = config.getString(HdfsSinkConnectorConfig.HDFS_URL_CONFIG);
+    hiveConf = new HiveConf();
+    String hiveConfDir = config.getString(HdfsSinkConnectorConfig.HIVE_CONF_DIR_CONFIG);
+    hiveConf.addResource(new Path(hiveConfDir, "hive-site.xml"));
+    SessionState.start(new CliSessionState(hiveConf));
+    cliDriver = new CliDriver();
+    hiveQLWriter = new HiveQLWriter(url, topicsDir);
+  }
+
+  /**
+   * @param tableName the name of the table to create in Hive.
+   * @param avroSchema the schema of the Avro files in HDFS.
+   */
+  public void createAvroTable(String tableName, Schema avroSchema) throws IOException {
+    // generate the HiveQL statements to run
+    String createAvroTableStmt = hiveQLWriter.createAvroTableStmt(tableName, avroSchema) + ";\n";
+    executeQuery(createAvroTableStmt);
+    log.info("Create Avro table complete.");
+  }
+
+  /**
+   * @param tableName the name of the table to create in Hive.
+   * @param avroSchema the schema of the Avro files in HDFS.
+   */
+  public void alterScheam(String tableName, Schema avroSchema) throws IOException {
+    String alterSchemaStmt = hiveQLWriter.alterSchemaStmt(tableName, avroSchema) + ";\n";
+    executeQuery(alterSchemaStmt);
+    log.info("Alter schema complete.");
+  }
+
+  /**
+   * @param tableName the name of the table to create in Hive.
+   * @param partition the partition to add to the table.
+   */
+  public void addPartition(String tableName, int partition) throws IOException {
+    String addPartitionStmt = hiveQLWriter.addPartitionStmt(tableName, partition) + ";\n";
+    executeQuery(addPartitionStmt);
+    log.info("Add partition complete.");
+  }
+  /**
+   * @param tableName the name of the table to create in Hive.
+   * @param partition the partition to add to the table.
+   */
+  public void setPartitionLoc(String tableName, int partition) throws IOException {
+    String setPartitionLocStmt = hiveQLWriter.setPartitionLocStmt(tableName, partition) + ";\n";
+    executeQuery(setPartitionLocStmt);
+    log.info("Set partition location complete.");
+  }
+
+  public void executeQuery(String query) throws IOException {
+    try {
+      log.debug("Using in-process Hive instance.");
+      String[] argv = getHiveArgs("-S", "-e", query);
+
+      int status = cliDriver.run(argv);
+      if (status != 0) {
+        throw new IOException("Hive CliDriver exited with status=" + status);
+      }
+    } catch (Exception e) {
+      throw new IOException("Exception thrown in Hive", e);
+    }
+  }
+
+  public String[] getHiveArgs(String... args) throws IOException {
+    List<String> newArgs = new LinkedList<>();
+    newArgs.addAll(Arrays.asList(args));
+    addHiveConfigs(hiveConf, conf);
```

Comment:
Why not just add these hive configs once? It seems like hiveConf and conf are always going to be the same since they aren't parameters to this method.